pkg/util/namespace.go

```diff
@@ -52,7 +52,10 @@ func DeleteNamespace(client kubeclient.Interface, namespace string) error {
 func EnsureNamespaceExist(client kubeclient.Interface, namespace string, dryRun bool) (*corev1.Namespace, error) {
 	namespaceObj := &corev1.Namespace{}
 	namespaceObj.Name = namespace
-
+	// It's necessary to set the label of namespace to make sure that the namespace is created by Karmada.
+	namespaceObj.Labels = map[string]string{
+		ManagedByKarmadaLabel: ManagedByKarmadaLabelValue,
+	}
```

Comment:
Note that the `EnsureNamespaceExist` is a common function, it shouldn't add labels inside of it.

Reply:
cc @XiShanYongYe-Chang

Reply:
How can we elegantly handle this label? By adding parameters?

Reply:
Something like that, we can have a dedicated function that accepts expected labels.

Reply:
How about adding a new func like this `EnsureNamespaceExistWithLabels`

Reply:
Hi @zhy76, we can add a new `EnsureNamespaceExistWithLabels` func to replace the old func call.

Reply:
OK 😄